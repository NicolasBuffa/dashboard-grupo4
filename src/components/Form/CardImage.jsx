import React from "react";

function CardImage({ image }) {
  return (
    <div className="productView-container_actualImgDetail">
      <div className="picture">
        {/* <p>{image}</p> */}
        {/* <img src={image} alt="" /> */}
      </div>

      <button className="buttonProductView">Quitar </button>
    </div>
  );
}

export default CardImage;
