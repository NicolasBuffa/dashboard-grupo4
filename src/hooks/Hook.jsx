import React from 'react'

function Hook() {
  return (
    <div>Hook</div>
  )
}

export default Hook