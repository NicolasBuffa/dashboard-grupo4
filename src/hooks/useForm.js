import { useState, useRef } from "react";
import { useParams } from "react-router-dom";
import { useNavigate } from "react-router-dom";

export const useForm = (initialForm) => {
  const navigate = useNavigate();
  const [formState, setFormState] = useState(initialForm);
  const images = useRef(null);
  const params = useParams();

  const onInputChange = ({ target }) => {
    const { name, value } = target;
    if (name === "price" || name === "stock") {
      setFormState({ ...formState, [name]: Number(value) });
    } else {
      setFormState({ ...formState, [name]: value });
    }
  };

  const handleIncrement = ({ target }) => {
    setFormState({ ...formState, stock: formState.stock + 1 });
  };
  const handleDecrement = ({ target }) => {
    if (formState.stock >= 0) {
      setFormState({ ...formState, stock: formState.stock - 1 });
    }
  };

  const handdleAddImage = ({ target }) => {
    // let aux = formState.images;
    // aux.push(target.value);
    setFormState({ ...formState, images: [...formState.images, target.value] });
  };

  const handleRemoveImage = (index) => {
    let aux = formState.images;
    aux.splice(index, 1);
    setFormState({ ...formState, images: aux });
  };
  const onResetForm = () => {
    setFormState(initialForm);
  };
  const handleSubmitNewProduct = async (ev) => {
    ev.preventDefault();

    fetch("http://localhost:4000/api/product", {
      method: "POST",
      body: JSON.stringify({
        ...formState,
      }),
      headers: {
        "Content-type": "application/json; charset=UTF-8",
      },
    })
      // .then((response) => response.json())
      // .then((json) => console.log(json));
      .then((response) =>{return response.status})
        .catch((error) => { console.error(error); });
  };
  const handleSubmit = async (ev) => {
    ev.preventDefault();
    fetch("http://localhost:4000/api/product", {
      method: "PUT",
      body: JSON.stringify({
        ...formState,
      }),
      headers: {
        "Content-type": "application/json; charset=UTF-8",
      },
    })
      // .then((response) => response.json())
      // .then((json) => console.log(json))
      .then((response) =>
        response.status === 200
          ? navigate("/products")
          : console.log(response.status)
      );
  };
  const handdleDelete = async (ev) => {
    ev.preventDefault();
    fetch(`http://localhost:4000/api/product?id=${params.id}`, {
      method: "DELETE",
      headers: {
        "Content-type": "application/json",
      },
    }).then((response) =>
      response.status === 200
        ? navigate("/products")
        : console.log(response.status)
    );
  };
  return {
    ...formState,
    formState,
    setFormState,
    onInputChange,
    handleSubmit,
    handdleDelete,
    handdleAddImage,
    handleRemoveImage,
    handleDecrement,
    handleIncrement,
    handleSubmitNewProduct,
    onResetForm,
  };
};
