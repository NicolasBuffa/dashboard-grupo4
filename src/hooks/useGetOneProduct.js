import { useState, useEffect } from "react";
import { useParams } from "react-router-dom";

export const useGetOneProduct = (id, initialForm = {}) => {
  const [product, setProduct] = useState(initialForm);
  const params = useParams();

  // GET ONE PRODUCT HOOK
  const getProduct = async () => {
    const resp = await fetch(`http://localhost:4000/api/product/${id}`);
    const data = await resp.json();
    setProduct(data);
  };

  useEffect(() => {
    getProduct();
    return () => {};
  }, [params.id]);

  return {
    ...product,
    product,
  };
};
