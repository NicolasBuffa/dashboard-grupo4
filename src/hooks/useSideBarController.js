import { useState } from "react"

const useSideBarController = ( sideBarComponent, menuButtonClose) => {
  const [input, setInput] = useState('menuButton MenuButtonMobileDesactivate');


  const ButtonControler = () => {
    if (input === 'menuButton MenuButtonMobileDesactivate') {
      sideBarComponent.current.className = 'SideBar SideBarActive'
      setInput('menuButton MenuButtonMobileActive')
      menuButtonClose.current.className = 'menuButtonClose'
    }
  }

  const outMenu = () => {

    if (input === 'menuButton MenuButtonMobileActive') {
      sideBarComponent.current.className = 'SideBar SideBarDesactivate'
      setInput('menuButton MenuButtonMobileDesactivate')
      menuButtonClose.current.className = 'menuButtonCloseUp'
    }
  }
  return (
    { input, ButtonControler, outMenu }
  )
}
export default useSideBarController