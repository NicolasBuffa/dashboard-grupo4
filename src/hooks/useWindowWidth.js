import { useState, useEffect } from "react";

//This hooks returns de width of the screen.
function useWindowWidth() {
  const [width, setWidth] = useState(window.innerWidth);
  useEffect(() => {
    const handleResize = () => {
      setWidth(window.innerWidth);
    };
    window.addEventListener("resize", handleResize);
    return () => {
      window.removeEventListener("resize", handleResize);
    };
  }, []);
  return width;
}

export default useWindowWidth;
