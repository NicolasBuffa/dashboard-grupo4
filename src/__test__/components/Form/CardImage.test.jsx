import CardImage from "../../../components/Form/CardImage";
import { render, screen } from "@testing-library/react";
import { MemoryRouter, useHref } from "react-router-dom";
import { formState, setFormState, image } from "..//..//..//hooks/useForm";
describe("TEST CardImage", () => {
  test("should first", () => {
    const { container } = render(
      <CardImage
        image={image}
        formState={formState}
        setFormState={setFormState}
      />,
      { wrapper: MemoryRouter }
    );
    expect(container).toMatchSnapshot();
  });
});
