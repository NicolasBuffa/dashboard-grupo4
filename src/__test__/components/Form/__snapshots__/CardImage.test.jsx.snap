// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST CardImage should first 1`] = `
<div>
  <div
    class="productView-container_actualImgDetail"
  >
    <div
      class="picture"
    >
      <img
        alt=""
      />
    </div>
    <button
      class="buttonProductView"
    >
      Quitar
       
    </button>
  </div>
</div>
`;
